tools/credentials.go

```diff
@@ -18,3 +18,13 @@ func ToCredentialsSecretData(credentials any) (map[string][]byte, error) {
 		CredentialsSecretKey: credentialBytes,
 	}, nil
 }
+
+func ToDecodedSecretDataCredentials(data map[string][]byte) (map[string]any, error) {
```

Comment:
Maybe `FromCredentialsSecretData` would be a better name as it would be the reverse of `ToCredentialsSecretData`